### PR TITLE
[TEST] parse global parameters from _common.json

### PR DESCRIPTION
**Pull request by javanna**
Replace the hardcoded global parameters in the yaml test suite with parameters parsed from the newly added _common.json file.

Relates to #22569

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestClient.java [hunk 1]
```diff
@@ -108,7 +101,8 @@ public ClientYamlTestResponse callApi(String apiName, Map<String, String> params
             if (restApi.getPathParts().contains(entry.getKey())) {
                 pathParts.put(entry.getKey(), entry.getValue());
             } else {
-                if (restApi.getParams().contains(entry.getKey()) || ALWAYS_ACCEPTED_QUERY_STRING_PARAMS.contains(entry.getKey())) {
+                if (restApi.getParams().contains(entry.getKey()) || restSpec.isGlobalParameter(entry.getKey())
+                        || "ignore".equals(entry.getKey())) {
```
Comment (nik9000) [inline test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestClient.java]: May want to leave a comment explaining `ignore`. I don't know what it is and I expect you had to do some digging to figure it out.

File: test/framework/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestSpec.java [hunk 2]
```diff
@@ -57,6 +60,10 @@ public ClientYamlSuiteRestApi getApi(String api) {
         return restApiMap.values();
     }
 
+    public boolean isGlobalParameter(String param) {
```
Comment (nik9000) [inline test/framework/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestSpec.java]: Maybe javadoc too.

File: test/framework/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestSpec.java [hunk 3]
```diff
@@ -66,15 +73,39 @@ public static ClientYamlSuiteRestSpec parseFrom(FileSystem fileSystem, String op
...
                     try (XContentParser parser = JsonXContent.jsonXContent.createParser(NamedXContentRegistry.EMPTY, stream)) {
-                        ClientYamlSuiteRestApi restApi = restApiParser.parse(jsonFile.toString(), parser);
-                        String filename = jsonFile.getFileName().toString();
-                        String expectedApiName = filename.substring(0, filename.lastIndexOf('.'));
-                        if (restApi.getName().equals(expectedApiName) == false) {
-                            throw new IllegalArgumentException("found api [" + restApi.getName() + "] in [" + jsonFile.toString() + "]. " +
-                                    "Each api is expected to have the same name as the file that defines it.");
+                        if (filename.equals("_common.json")) {
+                            String currentFieldName = null;
+                            while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
```
Comment (nik9000) [inline test/framework/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestSpec.java]: I think it'd be easier to read if this were `ObjectParser` stuff.

File: test/framework/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestSpec.java [hunk 4]
```diff
@@ -66,15 +73,39 @@ public static ClientYamlSuiteRestSpec parseFrom(FileSystem fileSystem, String op
...
+                                if (parser.currentToken() == XContentParser.Token.FIELD_NAME) {
+                                    currentFieldName = parser.currentName();
+                                } else if (parser.currentToken() == XContentParser.Token.START_OBJECT
+                                        && "params".equals(currentFieldName)) {
+                                    while (parser.nextToken() == XContentParser.Token.FIELD_NAME) {
+                                        String param = parser.currentName();
+                                        if (restSpec.globalParameters.contains(param)) {
```
Comment (nik9000) [inline test/framework/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestSpec.java]: This looks like a good candidate for `XContentParser#map`. Not quite the same, but close enough?

**Review by nik9000 — APPROVED** [review]
Asked for some javadoc. If you have time using `ObjectParser` on the parsing is probably easier to read. At least it is easier for me to read.

File: test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestClient.java (hunk 1 above)
Comment (javanna) [inline test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestClient.java]: not really, it was already part of the previous constant... :) I will add a comment.

File: test/framework/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestSpec.java (hunk 4 above)
Comment (javanna) [inline test/framework/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestSpec.java]: I don't want to parse everything, only the name of the object and skip the content.

File: test/framework/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestSpec.java (hunk 3 above)
Comment (javanna) [inline test/framework/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestSpec.java]: I simply took what we already had in RestApiParser and adapted it. I unfortunately do not find ObjectParsers easier to read also, sorry!

**Comment by javanna** [comment]
@nik9000 I pushed a new commit

File: test/framework/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestSpec.java (hunk 3 above)
Comment (nik9000) [inline test/framework/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestSpec.java]: It is all good!

**Comment by nik9000** [comment]
LGTM

**Pull request merged**